src/d3a/models/market/__init__.py

```diff
@@ -123,7 +123,7 @@ def _update_stats_after_trade(self, trade, offer, buyer, already_tracked=False):
         self._update_accumulated_trade_price_energy(trade)
         self.traded_energy = add_or_create_key(self.traded_energy, offer.seller, offer.energy)
         self.traded_energy = subtract_or_create_key(self.traded_energy, buyer, offer.energy)
-        self._update_min_max_avg_trade_prices(offer.price / offer.energy)
+        self._update_min_max_avg_trade_prices(offer.energy_rate)
```

Comment:
Again, out of curiosity: Do you know how much resources are saved when one-time precalculating the offer rate in the offer instance  instead of doing it here all the time?

Reply:
3-4 seconds on a total of 67 seconds in one tick, in a 150-house simulation, which is not extreme, but contributes to the slowdown.